### ISSUE TITLE
Add an option to disable the download progress log message

### ISSUE THREAD

**Issue by ChristianAylward** [streamlink/streamlink#1770]
### Checklist

- [ ] This is a bug report.
- [x] This is a feature request.
- [ ] This is a plugin (improvement) request.
- [x] I have read the contribution guidelines.

### Description

Apart from regular console use, I also run Streamlink from Java (through a Java Process) and redirect the system output & error streams to Java. If I tell Streamlink to use a player, all is fine. However, if I tell Streamlink to save the file, it stops writing to the file after less than 1 minute. It appears to still be capturing as there are no new log messages to indicate an error but the file doesn't get any bigger.

After some investigating, it turns out to be the "[download] ??? Written ???" log message that clears the current line in order to print the new value that is causing the problem (the print_inplace function in progress.py). That log message isn't printed in Java which was a clue to the problem.

I've tried the none loglevel option but the download progress log message still appears. My only option ATM without modifying the Streamlink code myself is to turn the loglevel up to debug level causing no line clearing, just a whole lot of logspam, BUT a successful recording.

TLDR;
When I specify -Q option for quiet logging, can it NOT print the download progress as well please.
Or maybe an extra CLI argument to disable the download progress message altogether (or disable current line clearing and just write the new value on a new line). Ideally, I'd still like to get the error/warning/info log messages and just disable the constantly updating download progress.

### Comments, screenshots, etc.

Thanks to all the contributors for this awesome software.

**Comment by beardypig** [comment]
I think the best thing here would be to only show that progress counter when there is a valid terminal to write to (can also remove it when using -q). Then you can set the log level to warning and still get those messages.

**Comment by luponata** [comment]
Same problem with python

**Comment by beardypig** [comment]
@ChristianAylward @luponata I have created a PR that should resolve the issues with this, sorry for the delay :)

**Comment by bastimeyer** [comment]
> after redirecting
> i found that the "[download] written" doesn't appear anymore

https://github.com/streamlink/streamlink/blob/bcdbfd0a8267663d11057511bc09de54afc90128/src/streamlink_cli/main.py#L959-L964

**Comment by luponata** [comment]
> > after redirecting
> > i found that the "[download] written" doesn't appear anymore
> 
> [streamlink/src/streamlink_cli/main.py](https://github.com/streamlink/streamlink/blob/bcdbfd0a8267663d11057511bc09de54afc90128/src/streamlink_cli/main.py#L959-L964)
> 
> Lines 959 to 964 in [bcdbfd0](/streamlink/streamlink/commit/bcdbfd0a8267663d11057511bc09de54afc90128)
> 
>  # Console output should be on stderr if we are outputting 
>  # a stream to stdout. 
>  if args.stdout or args.output == "-": 
>      console_out = sys.stderr 
>  else: 
>      console_out = sys.stdout

yes, seems legit

so IMHO, only if a stream is redirected to the stdout the --quiet option should apply also to the stderr
